### PR TITLE
Remove logs factories and merge with normal factories

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: component/receiver.go [hunk 1]
```diff
@@ -109,11 +109,6 @@ type ReceiverFactory interface {
 	// error will be returned instead.
 	CreateMetricsReceiver(ctx context.Context, params ReceiverCreateParams,
 		cfg configmodels.Receiver, nextConsumer consumer.MetricsConsumer) (MetricsReceiver, error)
-}
-
-// LogsReceiverFactory can create a LogsReceiver.
-type LogsReceiverFactory interface {
```
Comment (tigrannajaryan) [inline component/receiver.go:115]: I am not sure I like this approach more than having separate factories. This forces the factory to implement all 3 functions and return ErrDataTypeIsNotSupported from some of them.
I think the current approach is nicer for component implementors. In fact, we should probably consider splitting Trace and Metric factory similarly.
What do you think?

Comment (bogdandrutu) [reply component/receiver.go:115]: The helper factories do everything for you. I think we should not have too many factories as per https://github.com/open-telemetry/opentelemetry-collector/issues/1289

If you see the helper does all the return `ErrDataTypeIsNotSupported` and everything :)

Comment (bogdandrutu) [reply component/receiver.go:115]: See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/745 as a proof of using helper factories

Comment (bogdandrutu) [reply component/receiver.go:115]: Also we can always go with the other option if you want, but we will have:
* BaseFactory
* ReceiverTraces
* ReceiverMetrics
* ReceiverLogs
* ProcessorTraces
* ProcessorMetrics
* ProcessorLogs
* ExporterTraces
* ExporterMetrics
* ExporterLogs

Comment (tigrannajaryan) [reply component/receiver.go:115]: OK, I am not completely sure it is a better approach. With separate interfaces the helper would not be needed at all. Not a strong opinion, works this way too.

**Pull request merged**